src/AudioPlayer.ts

```diff
@@ -224,9 +240,6 @@ export default class AudioPlayer extends EventTarget {
     this._audioNode.start();
 
     if (this._audioElement.srcObject) {
-      this.addEventListener('ended', () => {
```

Comment:
Does this have any memory implications?

Reply:
Yes, if the consumer of this player doesn't properly destroy it using the `set srcObject(srcObject: ISourceObject)` above. This is the same behavior when using a regular HTML Audio element where srcObject needs to bet set to null to destroy it.